plugins/aggregators/histogram/histogram.go

```diff
@@ -151,6 +161,10 @@ func (h *HistogramAggregator) Add(in telegraf.Metric) {
 				index := sort.SearchFloat64s(buckets, value)
 				agr.histogramCollection[field][index]++
 			}
+			if h.ExpirationInterval != 0 {
+				addTime := timeNow()
```

Comment:
Are you aware that this changes your behavior slightly? Before, all metrics had the same `addTime`, now that time differs slightly between iterations in the for-loop. Just wanted to make you aware...

Reply:
Missed it. But after the discussion above about `time.Now()` performance, I think we can just add `addTime := timeNow()` to the start of this method.